src/directives/formly-form.js

```diff
@@ -366,6 +368,9 @@ function formlyForm(formlyUsability, formlyWarn, $parse, formlyConfig, $interpol
     function getFormlyFieldLikeLocals(field, index) {
       // this makes it closer to what a regular formlyExpression would be
       return {
+        // when field.model is not yet an object, this call should be evaluated in $scope.model context
+        // because this is a call where field.model gets resolved
+        model: angular.isObject(field.model) ? field.model : $scope.model,
```

Comment:
I think we should actually put this logic in the place where the field model gets resolved rather than here because it could lead to unexpected results in other expressions.